modules/effects/src/effect_creator.ts

```diff
@@ -42,30 +44,25 @@ type ObservableReturnType<T> = T extends false
  */
 export function createEffect<
   C extends EffectConfig,
-  T extends DispatchType<C>,
-  O extends ObservableReturnType<T>,
-  R extends O | ((...args: any[]) => O)
->(source: () => R, config?: Partial<C>): R {
+  DT extends DispatchType<C>,
+  OT extends ObservableType<DT, OT>,
+  R extends Observable<OT> | ((...args: any[]) => Observable<OT>)
+>(source: () => R, config?: Partial<C>): R & CreateEffectMetadata {
```

Comment:
I'm wondering if `CreateEffectMetadata` might confuse people if they look at the typings.

Reply:
That's what `createEffect` returns though. I think it's fine to provide that data.